### PR TITLE
Add rpc requests article

### PR DESCRIPTION
**Pull request by AlexTheLion123**
This PR aims to add an article detailing how to make RPC requests in your event handler. 

### Article details

The article largely follows the format of the [IPFS tutorial](https://docs.envio.dev/docs/HyperIndex/ipfs) and demonstrates how to get ERC20 token details inside an event handler in Envio

### Indexer details

It uses a single event handler for uniswap v3 pool created event. Inside the event handler, it fetches the name, symbol, and decimals of each token and caches them. See [here](https://github.com/AlexTheLion123/envio-uniswap-factory-indexer) for the ongoing progress of the indexer.

### CODE REVIEW

File: docs/HyperIndex/Guides/rpc-requests.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+In order to do this, we need to make RPC requests to the token contract.
+
+### Prerequisites
+
+This guide assumes basic familiarity with the viem library for making contract calls. Check out the [viem documentation](https://viem.sh/) for more information. This [medium article](https://medium.com/@0xape/typescript-and-viem-quickstart-for-blockchain-scripting-3f1846970b6f) is highly recommended for a gentle introduction to viem with a very similar example to what we are doing here.
+
+### Part 1: Create our Uniswap V3 `poolcreated` indexer
+
+`npx envio init`
+Contract address: `0x1F98431c8aD98523631AE4a59f267346ea31F984`
```
Comment (DenhamPreen) [inline docs/HyperIndex/Guides/rpc-requests.md:34]: At this point it's not entirely apparent which network this is. 
```suggestion
Ethereum mainnet contract address: `0x1F98431c8aD98523631AE4a59f267346ea31F984`
```
Could also simply hyperlink the address to an explorer - here's a pretty cool explorer built on top of envio
https://scope.sh/1/address/0x1F98431c8aD98523631AE4a59f267346ea31F984

File: docs/HyperIndex/Guides/rpc-requests.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+```typescript
+import {
+  UniswapV3Factory,
+  Pool,
+} from "generated";
+import { getTokenDetails } from "./tokenDetails";
+
+UniswapV3Factory.PoolCreated.handler(async ({ event, context }) => {
+  const entity: Pool = {
+    id: `${event.chainId}_${event.block.number}_${event.logIndex}`,
```
Comment (DenhamPreen) [inline docs/HyperIndex/Guides/rpc-requests.md:94]: I usually use dashes instead of underscores, both are perfectly fine just might be confusing for some users so I think in general we should try stick to one. I couldn't find anything now in the docs with one or the other but I know all the examples I've done would have '-'
```suggestion
    id: `${event.chainId}-${event.block.number}-${event.logIndex}`,
```

File: docs/HyperIndex/Guides/rpc-requests.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+      symbol: symbol0,
+      decimals: decimals0
+    })
+  } catch (error) {
+    console.log(error)
+    return
+  }
+
+  try {
+    const {name: name1, symbol: symbol1, decimals: decimals1} = await getTokenDetails(event.params.token0, event.chainId);
```
Comment (DenhamPreen) [inline docs/HyperIndex/Guides/rpc-requests.md:118]: I think this should be a 1?
```suggestion
    const {name: name1, symbol: symbol1, decimals: decimals1} = await getTokenDetails(event.params.token1, event.chainId);
```

File: docs/HyperIndex/Guides/rpc-requests.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  }
+});
+
+```
+
+### Part 2: Fetch token information
+
+Note how we use the `multicall` feature from Viem to batch our 3 queries into one request to improve performance and help avoid rate limiting. Very useful!
+
+Here, we use an alchemy API key to make RPC requests to the ethereum mainnet. You can get an alchemy API key [here](https://www.alchemy.com/) and add it to your `.env` file. Alternatively, use any other rpc provider of your choice.
```
Comment (DenhamPreen) [inline docs/HyperIndex/Guides/rpc-requests.md:137]: Alchemy is definitely a good robust option, might be worth linking to chainlist.org for finding some alternative free rpc providers

Comment (DenhamPreen) [reply docs/HyperIndex/Guides/rpc-requests.md:137]: In fact since we partner with dRPC could you rather mention them :)

File: docs/HyperIndex/Guides/rpc-requests.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+> src/tokenDetails.ts
+
+```typescript
+import { createPublicClient, getContract, http } from 'viem'
+import { mainnet } from 'viem/chains'
+import { Cache, CacheCategory } from "./cache";
+import { ERC20ABI } from './constants';
+
+const apiKey = process.env.ALCHEMY_API_KEY;
```
Comment (DenhamPreen) [inline docs/HyperIndex/Guides/rpc-requests.md:147]: I suggest rather making this the full rpc endpoint than an api key as it's more flexible to switch between providers. ie something like
```suggestion
const apiKey = process.env.RPC_URL;
```

**Comment by DenhamPreen** [comment]
Hey @AlexTheLion123, this is fantastic :) Thanks for writing this. 
The vercel deployment is failing as you don't have a seat so I'll make a superfluous commit into this repo to action the deployment. 
I've left a few small comments, but we should be able to get this published soon!

File: docs/HyperIndex/Guides/rpc-requests.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
+---
+id: rpc-requests
+title: RPC Requests
+sidebar_label: RPC Requests
+slug: /rpc-requests
+---
+
+# RPC Requests guide
+
+TLDR; The repo for the code base can be found [here](todo_add_link)
```
Comment (DenhamPreen) [inline docs/HyperIndex/Guides/rpc-requests.md:10]: Let's set it to this, and when we transfer the repo we can make sure it matches the below :)
```suggestion
TLDR; The repo for the code base can be found [here](https://github.com/enviodev/rpc-token-data-example)
```

**Comment by DenhamPreen** [comment]
Oh nvm I can't do a commit into your fork. However I could trigger it from the vercel ui
[article](https://envio-docs-o6mjay1n3-envio-9ec2d675.vercel.app/)

**Comment by DenhamPreen** [comment]
My last thought is lets adjust the name from rpc-guide to contract-state, since its a guide unique to fetching contract state using rpc. It might confuse some users that are trying to index events using rpc as a source.

**Review by DenhamPreen — APPROVED** [review]
